cf/userpoolclient.yaml

```diff
@@ -27,8 +27,11 @@ Resources:
       UserPoolId:
         Fn::ImportValue: !Sub "biomage-user-pool-case-insensitive-${Environment}::UserPoolId"
       CallbackURLs:
-        - !If [isProd, "https://scp.biomage.net/data-management",
-          !Sub "https://ui-${SandboxID}.scp-${Environment}.biomage.net/data-management"]
+        - !Sub ["${BaseUrl}/data-management", BaseUrl:
+            !If [ isProd,
+              !Sub ["https://${DomainName}", DomainName: !Join ["", [ Fn::ImportValue: !Sub "DomainName-${Environment}" ]]],
+              !Sub ["https://ui-${SandboxID}.${DomainName}", DomainName: !Join ["", [ Fn::ImportValue: !Sub "DomainName-${Environment}" ]]]
+          ]]
```

Comment:
same here -- better define a parameter for BaseUrl

Reply:
ah that's annoying

Reply:
is there any other place (non parameters place) where we could define this BaseUrl variable?

Reply:
I don't think there's an easy way to do this right now. CloudFormation allows for the creation and definition of [CustomResources](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/template-custom-resources.html), and according to [this SO post](https://serverfault.com/a/812040), with the help of a lambda, we can use this to make "variables". It sounds like a workaround though and might be too much for the scope that we have and functionality that we want.